packages/ifpack2/src/Ifpack2_BlockTriDiContainer_impl.hpp

```diff
@@ -867,6 +928,40 @@ namespace Ifpack2 {
       const local_ordinal_type A_n_lclrows = A->getLocalNumRows();
       const local_ordinal_type nparts = jacobi ? A_n_lclrows : partitions.size();
 
+      typedef std::pair<local_ordinal_type,local_ordinal_type> size_idx_pair_type;
+      std::vector<size_idx_pair_type> partsz(nparts);
+
+      if (!jacobi) {
+        for (local_ordinal_type i=0;i<nparts;++i)
+          partsz[i] = size_idx_pair_type(partitions[i].size(), i);
+        std::sort(partsz.begin(), partsz.end(),
+                  [] (const size_idx_pair_type& x, const size_idx_pair_type& y) {
+                    return x.first > y.first;
+                  });
+      }
+
+      local_ordinal_type n_subparts_per_part;
+      if (n_subparts_per_part_in == -1) {
+        // If the number of subparts is set to -1, the user let the algorithm
+        // decides the value automatically
+        using execution_space = typename impl_type::execution_space;
+
+        const int line_length = partsz[0].first;
+
+        const local_ordinal_type team_size = 
+          SolveTridiagsDefaultModeAndAlgo<typename execution_space::memory_space>::
+          recommended_team_size(blocksize, vector_length, internal_vector_length);
+
+        const local_ordinal_type num_teams = execution_space().concurrency() / team_size;
```

Comment:
```suggestion
        const local_ordinal_type num_teams = execution_space().concurrency() / (team_size * vector_length);
```
Assuming this is to estimate the maximum number of teams executing at once, it should account for the vector length